pyblish_maya/lib.py

```diff
@@ -331,10 +330,17 @@ def dock(window):
     if not main_window:
         raise ValueError("Could not find the main Maya window.")
 
-    self._dock_count += 1
-    dock = Dock(self._dock_count, parent=main_window)
+    dock = Dock(parent=main_window)
 
-    cmds.dockControl(label=window.windowTitle(), area="right", visible=True,
-                     content=dock.objectName(), allowedArea=["right", "left"])
+    dock_control = cmds.dockControl(label=window.windowTitle(), area="right",
+                                    visible=True, content=dock.objectName(),
+                                    allowedArea=["right", "left"])
 
     dock.layout().addWidget(window)
+
+    # delete any existing dock control
+    if self._dock_control:
+        if cmds.dockControl(self._dock_control, q=True, ex=True):
```

Comment:
For our future selves, when we've forgotten the details of `dockControl`, could you use the fully qualified names of arguments? E.g. I'm assuming `q=True` means `query=True`.